### PR TITLE
Repository pattern edited

### PR DESCRIPTION
**Pull request by vasiliv**
Added Async to method names

### CODE REVIEW

File: src/Infrastructure/UnitOfWork/UnitOfWork.cs [hunk 1]
```diff
@@ -16,14 +16,26 @@ public UnitOfWork(ApplicationDbContext context)
         {
             _context = context;
             AppUsers = new AppUserRepository(_context);
+            Maps = new MapRepository(_context);
```
Comment (svadrif) [inline src/Infrastructure/UnitOfWork/UnitOfWork.cs:19]: you should use interfaces in the constructor

File: src/Infrastructure/Repositories/AppUserRepository.cs [hunk 2]
```diff
@@ -11,9 +12,30 @@ namespace Infrastructure.Repositories
...
+        public override async Task<List<AppUser>> GetAllAsync()
         {
+            return await _context.AppUsers.AsNoTracking()
+                .OrderBy(b => b.LastName)
+                .ToListAsync();
+        }
 
+        public override async Task<AppUser> GetByIdAsync(Guid id)
+        {
+            return await _context.AppUsers.AsNoTracking()
```
Comment (svadrif) [inline src/Infrastructure/Repositories/AppUserRepository.cs:26]: here u can remove where and firstordefault then put .FindAsync(id) method

**Review by DoniyorJN — APPROVED** [review]

**Review by DoniyorJN — CHANGES_REQUESTED** [review]
Please resolve the conflicts

File: src/Infrastructure/Repositories/AppUserRepository.cs (hunk 2 above)
Comment (vasiliv) [reply src/Infrastructure/Repositories/AppUserRepository.cs]: We deleted this overridden method

Comment (vasiliv) [reply src/Infrastructure/Repositories/AppUserRepository.cs]: Changes done in Generic Repository

**Review by DoniyorJN — APPROVED** [review]

**Review by DoniyorJN — APPROVED** [review]
looks good

**Pull request merged**